webapp/src/ts/services/android-api.service.ts

```diff
@@ -0,0 +1,268 @@
+      return true;
+    }
+
+    if (routeSnapshot.params.id) {
+      this.router.navigate(['/', routeSnapshot.parent.routeConfig.path]);
+      return true;
+    }
+
+    // If we're viewing a tab, but not the primary tab, go to primary tab
+    const primaryTab = $('.header .tabs').find('> a:visible:first');
+    if (!primaryTab.is('.selected')) {
+      const href = primaryTab.attr('href');
+      if (href) {
+        this.router.navigate([href.replace('#','')]);
+        return true;
+      } else {
+        const message = 'Attempt to back to an undefined state [AndroidApi.back()]';
+        this.feedbackService.submit(message).catch(err => {
+          console.error('Error saving feedback', err);
+        });
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Handle the response from the MRDT app
+   * @param response The stringified JSON response from the MRDT app.
+   */
+  mrdtResponse(response) {
+    try {
+      this.mrdtService.respond(JSON.parse(response));
+    } catch(e) {
+      return console.error(
+        new Error(`Unable to parse JSON response from android app: "${response}", error message: "${e.message}"`)
+      );
+    }
+  }
+
+  /**
+   * Handle the response from the MRDT app
+   * @param response The stringified JSON response from the MRDT app.
+   */
+  mrdtTimeTakenResponse(response) {
+    try {
+      this.mrdtService.respondTimeTaken(JSON.parse(response));
+    } catch(e) {
+      return console.error(
+        new Error(`Unable to parse JSON response from android app: "${response}", error message: "${e.message}"`)
+      );
+    }
+  }
+
+  /**
+   * Handle the response from the simprints device
+   *
+   * @param requestType Indicates the response handler to call. Either 'identify' or 'register'.
+   * @param requestIdString The unique ID of the request to the simprints device.
+   * @param response The stringified JSON response from the simprints device.
+   */
+  simprintsResponse(requestType, requestIdString, response) {
+    const requestId = parseInt(requestIdString, 10);
+    if (isNaN(requestId)) {
+      return console.error(new Error('Unable to parse requestId: "' + requestIdString + '"'));
+    }
+    try {
+      response = JSON.parse(response);
+    } catch(e) {
+      return console.error(new Error('Unable to parse JSON response from android app: "' + response + '"'));
+    }
+    if (requestType === 'identify') {
+      //Simprints.identifyResponse(requestId, response);
+    } else if (requestType === 'register') {
+      //Simprints.registerResponse(requestId, response);
+    } else {
+      return console.error(new Error('Unknown request type: "' + requestType + '"'));
+    }
+  }
+
+  smsStatusUpdate(id, destination, content, status, detail) {
+    console.debug('smsStatusUpdate() :: ' +
+      ' id=' + id +
+      ', destination=' + destination +
+      ', content=' + content +
+      ', status=' + status +
+      ', detail=' + detail);
```

Comment:
```suggestion
    console.debug(`smsStatusUpdate() ::  id=${id}, destination=${destination}, content=${content}, status=${status}, detail=${detail}`);
```

I know this block was already formatted like this, but an alt-enter and "replace with template string" in webstorm makes it a bit more pleasant :)

Reply:
Unfortunately, if I do this, the line exceeds the allowed length. We don't have tslint enabled yet, so nothing complains, but it'll need reformatting when we enable it. 
For reference, these are our eslint rules: https://github.com/medic/eslint-config/blob/master/index.js#L12